relay/pkg/block/push.go

```diff
@@ -0,0 +1,214 @@
+package block
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/keep-network/tbtc/relay/pkg/btc"
+)
+
+// pullHeadersFromQueue waits until we have `headersBatchSize` headers from
+// the queue or until the queue fails to yield a header for
+// `headerTimeout` duration and returns them from the function.
+func (f *Forwarder) pullHeadersFromQueue(ctx context.Context) []*btc.Header {
```

Comment:
Can we separate pull functions to `pull.go`? I kinda like that we have a separate file for push logic instead of keeping it all in one big file but having a separation of push+pull+glue code would be even more awesome.

Reply:
`pull.go` is about pulling blocks from BTC chain and pushing them to queue (see https://github.com/keep-network/tbtc/pull/769). `push.go` is about pulling blocks from queue and pushing to the host chain. If we move `pullHeadersFromQueue` to `pull.go` we will make a small mess here.

Reply:
I like this idea! Do you mind I also leave a TODO for that and address it separately as I don't have `pull.go` in this PR?